eval.go

```diff
@@ -83,6 +83,10 @@ func (d *indexData) simplify(in query.Q) query.Q {
 			return d.simplifyMultiRepo(q, func(repo *Repository) bool {
 				return r.Set[repo.Name]
 			})
+		case *query.RepoIds:
```

Comment:
I found my changes worked for my use case without changes to `eval`. I think this is because the filtering is taking place in `shards.go` but I guess this logic is just to improve performance where we know ahead of time that we have all or no matches?